src/RenderLoop.cpp

```diff
@@ -269,69 +273,119 @@ void RenderLoop::KeyEvent(const SDL_KeyboardEvent& event, bool down)
+                             dest.makeDirectory();
+                             dest.setFileName(source.getFileName());
+                             
+                             Poco::File(source).copyTo(dest.toString());
+                             Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("Saved: " + source.getFileName()));
+                             poco_information_f1(_logger, "Saved preset to %s", dest.toString());
+                         } catch (Poco::Exception& e) {
+                             Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("Save failed: " + e.displayText()));
+                             poco_error_f1(_logger, "Failed to save preset: %s", e.displayText());
+                         }
+                         projectm_playlist_free_string(currentPreset);
+                     }
+                } else {
+                     Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("No save path configured."));
+                }
+            }
+            break;
+
+        case SDLK_D:
+            if (!modifierPressed) {
+                std::string savePath = Poco::Util::Application::instance().config().getString("application.savePresetsPath", "");
+                if (!savePath.empty()) {
+                     char* currentPreset = projectm_playlist_item(_playlistHandle, projectm_playlist_get_position(_playlistHandle));
+                     if (currentPreset) {
+                         try {
+                             Poco::Path source(currentPreset);
+                             Poco::Path dest(savePath);
+                             dest.makeDirectory();
+                             dest.setFileName(source.getFileName());
+                             
+                             Poco::File destFile(dest);
+                             if (destFile.exists()) {
+                                 destFile.remove();
+                                 Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("Deleted: " + source.getFileName()));
+                                 poco_information_f1(_logger, "Deleted preset from %s", dest.toString());
+                             } else {
+                                 Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("File not found in save dir."));
+                             }
+                         } catch (Poco::Exception& e) {
+                             Poco::NotificationCenter::defaultCenter().postNotification(new DisplayToastNotification("Delete failed: " + e.displayText()));
+                             poco_error_f1(_logger, "Failed to delete preset: %s", e.displayText());
+                         }
+                         projectm_playlist_free_string(currentPreset);
+                     }
+                }
+            }
```

Comment:
Same her,e please don't mix two different features into one PR.

Additionally, please keep the event handler code short. More than three lines of code or so should be implemented in a separate function and not within a switch case, otherwise the event handler will grow and become unmaintainable at some point.